lib/Sema/TypeCheckPattern.cpp

```diff
@@ -736,6 +736,74 @@ static bool validateParameterType(ParamDecl *decl, DeclContext *DC,
     }
     decl->getTypeLoc().setType(Ty);
   }
+  // If the param is not a 'let' and it is not an 'inout'.
+  // It must be a 'var'. Provide helpful diagnostics like a shadow copy
+  // in the function body to fix the 'var' attribute.
+  if (!decl->isLet() && !Ty->is<InOutType>()) {
+    auto innermostMethodCtx = dyn_cast_or_null<AbstractFunctionDecl>(DC);
+    if (!innermostMethodCtx || !innermostMethodCtx->hasBody()) {
+      // If there is not function body, just suggest removal of 'var'
+      TC.diagnose(decl->getLetVarInOutLoc(), diag::var_parameter_not_allowed)
+      .fixItRemove(decl->getLetVarInOutLoc());
```

Comment:
Is this really enough to fix SR-1020?  It seems like it would leave the space.  Maybe we need a ".fixItRemoveWithTrailingWhitespace" operation.  I'm sure that this comes up somewhere else in the compiler as well (grep for .fixItRemove and look for some code doing CharSourceRange gymnastics).